pkg/controller/controller.go

```diff
@@ -94,12 +98,13 @@ type ReQueue interface {
 }
 
 // NewQueueRetryManager safely creates a new QueueRetryManager.
-func NewQueueRetryManager(queue ReQueue, keyFn kcache.KeyFunc, retryFn RetryFunc) *QueueRetryManager {
+func NewQueueRetryManager(queue ReQueue, keyFn kcache.KeyFunc, retryFn RetryFunc, rate float32, burst int) *QueueRetryManager {
 	return &QueueRetryManager{
 		queue:     queue,
 		keyFunc:   keyFn,
 		retryFunc: retryFn,
 		retries:   make(map[string]int),
+		limiter:   kutil.NewTokenBucketRateLimiter(rate, burst),
```

Comment:
it also means any consumers of this now have to import, understand, and instantiate, another type instead of two primitives.  i'm not convinced that's an improvement from a usability+readability perspective.  I like that the RateLimiter interface is currently encapsulated so users of QueueRetryManager don't need to be aware of it, the potential benefits w/ respect to test code (that you mention below) not withstanding.